src/lava/lib/dl/netx/blocks/process.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier:  BSD-3-Clause
+
+"""Composable blocks processes in Lava."""
+
+from typing import Type, Union
+import numpy as np
+import h5py
+from lava.magma.core.process.ports.ports import InPort, OutPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.proc.dense.process import Dense as DenseSynapse
+from lava.proc.conv.process import Conv as ConvSynapse
+
+
+class AbstractBlock(AbstractProcess):
+    """Abstract block definition.
+
+    Parameters
+    ----------
+    shape : tuple or list
+        shape of the block output in (x, y, z) or WHC format.
+    neuron_params : dict
+        dictionary of neuron parameters. Defaults to None.
+    has_graded_input : dict
+        flag for graded spikes at input. Defaults to False.
+    """
```

Comment:
Do these docstring params need updating?

Reply:
Based on __init__ functionality

Reply:
If I haven't misunderstood something, AbstractBlock is the major abstraction introduced by netx, so the doc should be reasonably thorough, I think. Perhaps give some understanding of the purpose for this abstraction and how it relates to other key elements of Lava, like processes, compiler, or runtime. Especially since it's abstract, it could be easily misused by downstream APIs if the rationale isn't clear.

Also, do we strongly prefer AbstractBlock for this class over simply Block? Block is an uncommon enough name that I don't think there would be major name conflicts. E.g. the base process is simply Process, rather than AbstractProcess.

Reply:
Those are the three parameter it takes. Other members are paceholders which get properly initialized in the leaf Block classes.